### PR TITLE
Update carousels to use Swiper on hourofcode.com/beyond

### PR DESCRIPTION
**Pull request by kelbyhawn**
Updates the carousels on https://hourofcode.com/beyond to use Swiper.

## Links
Jira ticket: [ACQ-1885](https://codedotorg.atlassian.net/browse/ACQ-1885)

## Testing story
Tested locally

----

## Elementary 

https://github.com/code-dot-org/code-dot-org/assets/9256643/ebc14d51-e876-4786-9c01-415e5540ffa4

## Middle

https://github.com/code-dot-org/code-dot-org/assets/9256643/f48dbf5c-b6ba-4217-8e6a-72c4063b0bd8

## High 

https://github.com/code-dot-org/code-dot-org/assets/9256643/ffd368d5-b43c-4eb9-8694-e79cf2a77553

### CODE REVIEW

**Review by hannahbergam — APPROVED** [review]
I don't know that I have enough expertise to understand what I'd be looking for, but I appreciate the videos! If at all possible, I'd love to have a followup ticket to update the buttons on those cards to be more descriptive than 'Go', for accessibility purposes! I'm not sure what the screenreader experience is with swiper to begin with, but if those buttons don't have aria labels at least, it would be a good future work item<3

**Comment by kelbyhawn** [comment]
> If at all possible, I'd love to have a followup ticket to update the buttons on those cards to be more descriptive than 'Go'...

Swiper has pretty good accessibility built in navigation-wise ([see docs](https://swiperjs.com/swiper-api#accessibility-a11y)), but I made a [ticket](https://codedotorg.atlassian.net/browse/ACQ-1893) to add aria labels for the buttons 🙌 

https://github.com/code-dot-org/code-dot-org/assets/9256643/f1c2a667-e958-43ce-8041-defab4881109

**Comment by hannahbergam** [comment]
> > If at all possible, I'd love to have a followup ticket to update the buttons on those cards to be more descriptive than 'Go'...
> 
> Swiper has pretty good accessibility built in navigation-wise ([see docs](https://swiperjs.com/swiper-api#accessibility-a11y)), but I made a [ticket](https://codedotorg.atlassian.net/browse/ACQ-1893) to add aria labels for the buttons 🙌
> 
>  Screen.Recording.2024-05-16.at.1.05.07.PM.mov

Yesss this is what I was hoping to be true about Swiper. Even more important then for us to give more accurate link labels! Thank you!

File: apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// This installs the Swiper library WebComponent.
+// See docs here: https://swiperjs.com/swiper-api#web-component.
+
+// import function to register Swiper custom elements
+import {register} from 'swiper/element/bundle';
+// register Swiper custom elements
+register();
+
+const actionBlocksParams = {
```
Comment (bethanyaconnor) [inline apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js:11]: Do these change from page to page? Wondering if it's worth pulling this out into a shared file?

File: pegasus/sites.v3/hourofcode.com/views/course_carousels.haml [hunk 2]
```diff
@@ -18,10 +18,6 @@
 
   var ua = navigator.userAgent;
 
-  var is_safari4 = ua.indexOf("Safari/") != -1 && ua.indexOf("Version/4.") != -1 && ua.indexOf("Chrome") == -1 && ua.indexOf("Chromium") == -1;
-  var is_netscape7 = ua.indexOf("Netscape/7.") != -1;
-  var is_ie5 = ua.indexOf("MSIE 5.") != -1;
-
```
Comment (bethanyaconnor) [inline pegasus/sites.v3/hourofcode.com/views/course_carousels.haml:24]: thank you for removing these!

File: pegasus/sites.v3/hourofcode.com/views/course_carousel.haml [hunk 3]
```diff
@@ -12,35 +11,26 @@
...
-        .button-wrapper
-          %button.slide-prev{href: "#", style: "border:none; background-color: var(--neutral_light);"}
-            %img{src: "//code.org/images/carousel/prev.svg", alt: "previous"}
-          %button.slide-next{href: '#', style: "border:none; background-color: var(--neutral_light);"}
-            %img{src: "//code.org/images/carousel/next.svg", alt: "next"}
-
-      .pagination
-        %div{id: "pagination-#{id}"}
+    %button.swiper-nav-prev{class: id}
+    %button.swiper-nav-next{class: id}
```
Comment (bethanyaconnor) [inline pegasus/sites.v3/hourofcode.com/views/course_carousel.haml:36]: where is `id` coming from? (for my own information)

File: apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js (hunk 1 above)
Comment (kelbyhawn) [reply apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js:11]: I've thought about this, and refactoring how we can set up Swiper to be less repeat-y in general 🙌  Wanted to wait until everything was moved over and cleaned up though. I'll make a ticket for the backlog for this. 

**Edit:** here's the ticket [ACQ-1896](https://codedotorg.atlassian.net/browse/ACQ-1896)

File: pegasus/sites.v3/hourofcode.com/views/course_carousel.haml (hunk 3 above)
Comment (kelbyhawn) [reply pegasus/sites.v3/hourofcode.com/views/course_carousel.haml:36]: Tbh I don't know 😅 They show up magically in the DOM. I think this list might be coming from the tutorials somewhere, but I haven't looked into this code outside of making it work with Swiper.

File: apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/sites/hourofcode.com/pages/views/swiper_page_beyond.js:11]: thanks for making the ticket!

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**